jupyterhub/proxy.py

```diff
@@ -26,6 +28,54 @@
 from .utils import url_path_join
 
 
+class RouteSpec(namedtuple('RouteSpec', ['path', 'host'])):
+    """Object wrapping a proxy route specification
+
+    Attributes:
+
+        path (str): The URL path prefix for this route. Required.
+        host (str): The hostname used for routing when host-based routing is enabled.
+
+    """
+    def __new__(cls, path, *, host=''):
```

Comment:
IMO host should be None (or '*') rather than '' when we don't care. Some proxy systems might have special meaning for ''.